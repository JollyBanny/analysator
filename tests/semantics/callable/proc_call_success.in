procedure doSomething(d:double; int: integer; str: string; ch: char; bool: boolean);
begin
end;

begin
    doSomething(1.0, 1, 'aa', 'a', true);
end.