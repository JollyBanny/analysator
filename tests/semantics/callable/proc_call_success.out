program
└──── program block
      ├──── PROCEDURE
      │     ├──── header
      │     │     ├──── dosomething
      │     │     ├──── parameter
      │     │     │     ├──── d
      │     │     │     └──── double
      │     │     ├──── parameter
      │     │     │     ├──── int
      │     │     │     └──── integer
      │     │     ├──── parameter
      │     │     │     ├──── str
      │     │     │     └──── STRING
      │     │     ├──── parameter
      │     │     │     ├──── ch
      │     │     │     └──── char
      │     │     └──── parameter
      │     │           ├──── bool
      │     │           └──── boolean
      │     └──── block
      │           └──── Compound statement
      │                 └──── Empty statement
      └──── Compound statement
            └──── Call statement
                  └──── dosomething
                        ├──── 1
                        ├──── 1
                        ├──── 'aa'
                        ├──── 'a'
                        └──── TRUE

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  builtins                                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  integer                                                                                         │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  double                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  char                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  string                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  boolean                                                                                         │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  global                                                                                                              │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Procedure        │  dosomething                │  No return type                                                    │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
    ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
    │  dosomething params:                                                                                                 │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  d                          │  double                                                  │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  int                        │  integer                                                 │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  str                        │  string                                                  │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  ch                         │  char                                                    │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  bool                       │  boolean                                                 │         │
    └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
    ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
    │  dosomething locals:                                                                                                 │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  d                          │  double                                                  │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  int                        │  integer                                                 │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  str                        │  string                                                  │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  ch                         │  char                                                    │         │
    ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
    │  Parameter        │  bool                       │  boolean                                                 │         │
    └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
